doc/syntax/comments.rdoc

```diff
@@ -35,3 +35,188 @@ syntax error:
     Will not work
     =end
   end
+
+= Magic Comments
```

Comment:
I believe everything should be 1 level deeper (e.g. `== Magic Comments`, `=== Alternative syntax` etc.)